design-documents/graph-ql/coverage/search.md

```diff
@@ -0,0 +1,79 @@
+# Queries
+
+```graphql
+type Query {
+    multiSearch(phrase: String!, productSize: Int = 10): MultiSearchResponse!
+    #Filter supports multiple clauses which will be wrapped in logical AND operator
+    productSearch(phrase: String!, filter: [SearchClauseInput], sort: [ProductSearchSortInput]): ProductSearchResponse!
```

Comment:
right now advanced search does not provide facets. 
I think faceting should be optional and specified at request time

Reply:
Everything is optional in graphql, you need to explicitly request what you need. Also advanced search functionality is not currently covered by grapqhl and I doubt if we want to cover it.

Reply:
Everything in response is optional in graphql. Moreover, advanced search is not currently covered and I didn't plan to cover it in this iteration, because functionality itself is very questionable and repeats most of the regular search